js_modules/dagit/packages/eslint-config/rules/missing-graphql-variables-type.js

```diff
@@ -0,0 +1,96 @@
+const {ESLintUtils, AST_NODE_TYPES} = require('@typescript-eslint/utils');
+
+const fs = require("fs")
+const path = require("path")
+
+const createRule = ESLintUtils.RuleCreator((name) => name);
+
+/**
+ * Strategy:
+ *  1. Pass over all useQuery calls and append "Variables" to the identifierName of the first type
+ *    -- eg: useQuery<SomeQuery>() --> SomeQueryVariables
+ *  2. Check if "SomeQueryVariables" is exported from the file that "SomeQuery" is imported from.
+ *  3. If it's not then continue
+ *     If it is then check that the second type argument to useQuery is that SomeQueryVariables.
+ *     If not then throw an error.
+ *  
+ * TODO: Add auto-fixer
+ */
+
+module.exports = { 
+  rule: createRule({
+    create(context) {
+      return {
+        [AST_NODE_TYPES.CallExpression](node) {
```

Comment:
Wow this line broke my mental compiler - have never seen the name of the function supplied dynamically like this. 💯